src/back/services/discord.ts

```diff
@@ -133,7 +146,7 @@ export class DiscordService {
     if (user) {
       try {
         const profile = await getProfile(user)
-        const profileHasName = !!profile && !!profile.name && profile.name.length > 0
+        const profileHasName = !!profile && profile.hasClaimedName && !!profile.name && profile.name.length > 0
```

Comment:
(Not for this PR but we have this check here and in useProfile. We could add a function to reuse in both files)